src/lxml/readonlytree.pxi

```diff
@@ -290,6 +290,26 @@ cdef class _ReadOnlyElementProxy(_ReadOnlyProxy):
                     return funicode(self._c_node.ns.prefix)
             return None
 
+    property nsmap:
```

Comment:
Looks like a copy of `_Element.nsmap`, so the dict building should go into a shared helper function.